src/Vault.sol

```diff
@@ -118,14 +118,30 @@ contract Vault is IVault, VaultToken, Ownable {
     }
 
     /// @inheritdoc IVault
-    function settle(Currency currency) external payable override isLocked returns (uint256 paid) {
+    function settle(Currency currency) public payable override isLocked returns (uint256 paid) {
         uint256 reservesBefore = reservesOfVault[currency];
         reservesOfVault[currency] = currency.balanceOfSelf();
         paid = reservesOfVault[currency] - reservesBefore;
         // subtraction must be safe
         SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
     }
 
+    function settleAndRefund(Currency currency) external payable isLocked returns (uint256 paid, uint256 refund) {
```

Comment:
is it better to accept an extra param for example `address to` as the refund target

Reply:
added `to` in param 👍